CAN-D/Src/can.c

```diff
@@ -231,31 +232,46 @@ void APP_CAN_MonitorTask(void const* argument)
 {
     uint8_t usbTxCnt = 0;
     osEvent event;
-    CANRxMessage* msg;
+    CANRxMessage* canRxMsg;
+    size_t usbMaxMsgLen = CAN_USB_DATA_SZ_BYTES + 10; // Max length of the serialized data
+    uint8_t usbTxMsg[usbMaxMsgLen]; // Serialized (packaged) protobuf data
+    size_t usbTxNumBytes = 0; // Number of bytes in serialized data
+    FromEmbedded fromEmbeddedMsg = FromEmbedded_init_zero;
 
     for (;;) {
         /* This is just used to test the SD card functionality */
-        //  const uint8_t data[] = "YELLOW";
-        //  APP_FATFS_LogSD(data, 6, "CAN_data.log");
+        // const uint8_t data[] = "YELLOW";
+        // APP_FATFS_LogSD(data, 6, CAN_LOG_FILENAME);
 
         // Pend on any CAN Rx data
         event = osMessageGet(CANRxQueueHandle, 0);
         if (event.status == osEventMessage) {
-            msg = event.value.p;
+            canRxMsg = event.value.p;
             if (mAppConfiguration.SDStorage == APP_ENABLE) {
                 // Write data to SD card
-                APP_FATFS_LogSD((const uint8_t*)msg->data, 8, canLogIdentifier);
+                APP_FATFS_LogSD((const uint8_t*)canRxMsg->data, CAN_RX_MSG_DATA_SZ_BYTES, canLogIdentifier);
             }
 
+            // Pack the protobuf message
+            fromEmbeddedMsg.contents.canDataChunk.data.size = 8;
+            fromEmbeddedMsg.contents.canDataChunk.has_id = 1;
+            fromEmbeddedMsg.contents.canDataChunk.has_data = 1;
+            fromEmbeddedMsg.which_contents = 1;

```

Comment:
Should be `FromEmbedded_canDataChunk_tag`

Reply:
Thanks! Couldn't find this one.